barman/postgres.py

```diff
@@ -665,6 +665,43 @@ def has_checkpoint_privileges(self):
                 )
                 return None
 
+    @property
+    def has_monitoring_privileges(self):
+        """
+        Returns true if the current user is a superuser or if the user has the
+        necessary privileges to monitor system status.
+        """
+        if self.is_superuser:
+            return True
+        else:
+            monitoring_check_query = """
+            SELECT
+            (
+                pg_has_role(CURRENT_USER, 'pg_monitor', 'MEMBER')
+                OR
+                (
+                    pg_has_role(CURRENT_USER, 'pg_read_all_settings', 'MEMBER')
+                    AND pg_has_role(CURRENT_USER, 'pg_read_all_stats', 'MEMBER')
+                )
+            )
+            FROM
+            pg_user
+            WHERE
+            usename = CURRENT_USER
+            """.format(
+                **self.name_map
+            )
```

Comment:
```suggestion
            SELECT
            (
                pg_has_role(CURRENT_USER, 'pg_monitor', 'MEMBER')
                OR
                (
                    pg_has_role(CURRENT_USER, 'pg_read_all_settings', 'MEMBER')
                    AND pg_has_role(CURRENT_USER, 'pg_read_all_stats', 'MEMBER')
                )
            )
```

We are not using anything from the `pg_catalog.pg_user` view, so I guess we can remove that part of the query.
Similarly, we don't have anything to interpolate in the string.